File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        bracket = {")" : "(" , "}" : "{" , "]" : "["} 

        for ch in s:
            if ch in bracket:
                if not stack or stack[-1] != bracket[ch]:
                    return False
                stack.pop()
            else:
                stack.append(ch)
        
        return stack == []

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();

        char[] ch = s.toCharArray();

        for(char c : ch){

            if(c == '(' || c == '[' || c == '{'){
                stk.add(c);
            }
            else{
                if(c == ')'){
                    if(!stk.isEmpty() && stk.pop()== '(') continue;
                    else return false;
                }
                else if(c== ']'){
                    if(!stk.isEmpty() && stk.pop() == '[') continue;
                    else return false;
                }
                else if(c== '}'){
                    if(!stk.isEmpty() && stk.pop() == '{') continue;
                    else return false;
                }
            }
        }

        return stk.isEmpty();
    }
}